lib/widgets/compose_box.dart

```diff
@@ -1523,14 +1523,27 @@ class _ComposeBoxState extends State<ComposeBox> with PerAccountStoreAwareStateM
 
   @override
   void onNewStore() {
+    final newStore = PerAccountStoreWidget.of(context);
+
+    if (_controller == null) {
+      _setNewControllerForNarrow(newStore);
+      return;
+    }
+
+    switch (_controller) {
+      case StreamComposeBoxController():
+        (_controller as StreamComposeBoxController).topic.store = newStore;
```

Comment:
This can be simplified by extracting a local variable as a copy of the field, so that type promotion works:

```suggestion
    final controller = _controller;
    switch (controller) {
      case StreamComposeBoxController():
        controller.topic.store = newStore;
```